### DISCUSSION TITLE
Why not nosql file based db?

### DISCUSSION THREAD

**Discussion by michaldev** [fief-dev/fief#403] [Ideas]
I have returned to the project after a few months and noticed that the project manager has set new priorities. The focus now is on preparing the project for self-hosting options, including fast configuration on one's own server.

Have you considered using a NoSQL file-based database instead of MySQL? In my opinion, it would be sufficient because we only have simple data in the database. Configuring such a database is very easy. We have some existing options like TinyDB, DictDataBase, FileXdb.py, and many more.

Another advantage is that it does not burden the server as much as the mySQL database.

**Comment by frankie567** [comment]
Fief has been designed and written with relational databases in mind; I fear it would be very difficult to support a NoSQL database.

If you want to store Fief's data in a simple and file-contained format, you can use SQLite, we support that out-of-the-box: https://docs.fief.dev/self-hosting/environment-variables/#database
